libminifi/include/agent/agent_version.h

```diff
@@ -18,6 +18,8 @@
 #ifndef LIBMINIFI_INCLUDE_AGENT_AGENT_VERSION_H_
 #define LIBMINIFI_INCLUDE_AGENT_AGENT_VERSION_H_
 
+#undef COMPILER
+
```

Comment:
Why is this needed? Could you add a comment?

Reply:
TBH, I don't know why it's needed.  Something somewhere `#define`s `COMPILER` to the MSVC version, but I couldn't find what and where.

I'll take another look and add the comment if I find out, but I didn't think `// something somewhere #defines COMPILE to the MSVC version on Windows` would be a useful comment.

Reply:
That's fine, thank you.

Reply:
I have found the culprit (it's https://github.com/python/cpython/blob/3.10/PC/pyconfig.h) and added a comment.